### ISSUE TITLE
In terminal output, linebreaks are offset by the end of the previous line

### ISSUE THREAD

**Issue by nicogreenarry** [fish-shell/fish-shell#7218]
System details:
* fish, version 3.1.2
* macOS 10.15.4
  * Darwin Nicos-MBP 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
* TERM: xterm-256color
* When I ran `sh -c 'env HOME=$(mktemp -d) fish'`, I still observed the problematic behavior described below.

I just switched from zsh to fish. I installed it via `brew install fish`.

**The problem:** For every command, the start of one line of output is offset by the horizontal position of the end of the previous line of output. It makes multiline output very difficult to read. Here's an example of a few commands:

```
                                                           ⏎                                                                                                                                          ⋊> ~/d/triplebyte on nico/TB-2541-automate-screen-only-state ⨯ omf install beloglazov                                                                                                         12:28:06
Updating https://github.com/oh-my-fish/packages-main master... Done!
                                                                    Installing package beloglazov
                                                                                                 ✔ beloglazov successfully installed.
                                                                                                                                     ⏎                                                                12:28 
12:36 triplebyte (nico/TB-2541-automate-screen-only-state) +2 ✗ echo test; echo test; echo test
test
    test
        test
            ⏎
```

I'm happy to record an asciinema if anyone thinks it will help. I started with the above because I think it illustrates the issue pretty well.

Things I tried:
* I tried installing a different omf theme, in case it was a problem with the default theme, but that didn't fix the issue.
* I uninstalled oh-my-fish entirely, but still observed the same thing in vanilla fish.

Any thoughts on how to fix this?

**Comment by krobelus** [comment]
Sounds similar to #7209, so the output of `locale` and `stty -a` in `fish` and `zsh` could provide some insight.

**Comment by nicogreenarry** [comment]
```
> locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=

> stty -a
speed 0 baud; 57 rows; 198 columns;
lflags: -icanon -isig -iexten -echo -echoe -echok -echoke -echonl
-echoctl -echoprt -altwerase -noflsh -tostop -flusho -pendin
-nokerninfo -extproc
iflags: -istrip -icrnl -inlcr -igncr -ixon -ixoff -ixany -imaxbel -iutf8
-ignbrk -brkint -inpck -ignpar -parmrk
oflags: -opost -onlcr -oxtabs -onocr -onlret
cflags: -cread cs5 -parenb -parodd -hupcl -clocal -cstopb -crtscts
-dsrflow -dtrflow -mdmbuf
cchars: discard = ^@; dsusp = ^@; eof = ^@; eol = ^@; eol2 = ^@;
erase = ^@; intr = ^@; kill = ^@; lnext = ^@; min = 0; quit = ^@;
reprint = ^@; start = ^@; status = ^@; stop = ^@; susp = ^@;
time = 0; werase = ^@;
```

**Comment by nicogreenarry** [comment]
By the way, I'm now observing the exact same line-offsetting in zsh as I was in fish. I wasn't observing this before installing fish.

**Comment by faho** [comment]
Yeah, `-opost`. Something is breaking your terminal modes and fish isn't yet restoring them, fixed by #7133.

**Comment by nicogreenarry** [comment]
Excellent! That did the trick. Thanks for the quick diagnosis :)
